### PR TITLE
Document cyclical error in include in UG

### PR DESCRIPTION
**Pull request by yucheng11122017**
**What is the purpose of this pull request?**

- [X] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Documents #2177

**Overview of changes:**
Added a warning about the cyclical error in includes when variables have the same name. Included suggested solution for this bug.

**Anything you'd like to highlight/discuss:**

**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**
```Document cyclical error in include in UG```

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes

### CODE REVIEW

**Comment by damithc** [comment]
As this is a rare case (i.e., most users will not need to know about it), we should present it in a way most readers are not forced to read it but they can locate this information when they need to. Any ideas?
Perhaps a collapsed panel with a heading like `A caveat, if you are recursive includes ...` ?

**Comment by tlylt** [comment]
Thank you @yucheng11122017 for taking this up. Could we:
- add the explanation mentioned in https://github.com/MarkBind/markbind/issues/2177#issuecomment-1441293199 before the new panel to improve the current description, such that devs like yourself understand what's the rationale for such design
- construct an "MVP" example such that we show the cause, the error (which you have already included), and the resolution (which is missing)

File: docs/userGuide/syntax/includes.md [hunk 1]
```diff
@@ -155,6 +155,61 @@ You can also specify include variables within the `<include>` tag itself by addi
...
+        </include>
+      </span>
+    </include>
+  </span>
+</include>
+```
+
+To fix this problem, copy the content of the `src` file and replace the variables with the defined values as such:
+
+```html {.line-nums highlight-lines="3"}
```
Comment (tlylt) [inline docs/userGuide/syntax/includes.md:204]:
```suggestion
```html {.line-numbers highlight-lines="3"}
```

File: docs/userGuide/syntax/includes.md [hunk 2]
```diff
@@ -155,6 +155,61 @@ You can also specify include variables within the `<include>` tag itself by addi
...
+
+This is because the inner variable of the same name will be replaced with the outer variable, which contains the inner variable. 
+This inner variable is once again overridden to result in another inner variable and so on, causing a cyclical error.
+
+To fix this issue, do not use an inner `include` if they use the same variables.
+Instead, copy the content of the `src` file and replace the variables with the defined values.
+
+Example:
+
+```html
```
Comment (tlylt) [inline docs/userGuide/syntax/includes.md:172]:
```suggestion
```html {.line-numbers}
```

File: docs/userGuide/syntax/includes.md [hunk 3]
```diff
@@ -155,6 +155,61 @@ You can also specify include variables within the `<include>` tag itself by addi
 </div>
 
 If the same variable is defined in a chain of `<include>`s (e.g. `a.md` includes `b.md` includes `c.md`...), variables defined in the top-most `<include>` will take precedence. Global variables (`_markbind/variables.md`) will take precedence over any `<include>` variables.
+This is to allow the outer context to adapt the reused content without changing its actual content.
+
+<panel header="If you are using recursive includes...">
```
Comment (damithc) [inline docs/userGuide/syntax/includes.md:160]: collapse by default?

Comment (yucheng11122017) [reply docs/userGuide/syntax/includes.md:160]: Do you mean something like this @damithc?
Panels are collapsed by default unless `expanded` is set to true
image

Comment (damithc) [reply docs/userGuide/syntax/includes.md:160]: > Panels are collapsed by default unless `expanded` is set to true

@yucheng11122017  Ah, I forgot that. Ignore my comment in that case.
Feel free to use a better title for the panel BTW. The one I suggested was just from the top of my head ...

Comment (tlylt) [reply docs/userGuide/syntax/includes.md:160]: @yucheng11122017 how about changing the title to "Preventing cyclical errors when using multiple includes with same variables"?

**Pull request merged**